pytorch_lightning/trainer/trainer.py

```diff
@@ -942,8 +943,10 @@ def _run(self, model: "pl.LightningModule") -> Optional[Union[_EVALUATE_OUTPUT,
         if self.state.fn == TrainerFn.FITTING:
             self.call_hook("on_fit_end")
 
-        # teardown
-        self._call_teardown_hook(model)
+        # teardown if necessary (similar calls for spawn plugins are excluded as they have
+        # been included at the end of `new_process` functions)
+        if self._distrib_type not in DistributedType.interactive_compatible_types():
+            self._call_teardown_hook(model)
```

Comment:
I will follow up on this. I am convinced `DistributedType.interactive_compatible_types` is not the right choice for what we are trying to solve here as it contains too many plugins.